### PR TITLE
run clippy

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/cargo/ops/cargo_rustc/links.rs [hunk 1]
```diff
@@ -4,6 +4,7 @@ use core::PackageId;
 use util::CargoResult;
 use super::Unit;
 
+#[derive(Default)]
```
Comment (alexcrichton) [inline src/cargo/ops/cargo_rustc/links.rs]: This can be removed

File: src/cargo/ops/cargo_rustc/mod.rs [hunk 2]
```diff
@@ -533,12 +533,10 @@ fn build_base_args(cx: &Context,
     // exclusive.
     if unit.target.can_lto() && lto && !unit.target.for_host() {
         cmd.args(&["-C", "lto"]);
-    } else {
+    } else if let Some(n) = codegen_units {
         // There are some restrictions with LTO and codegen-units, so we
         // only add codegen units when LTO is not used.
```
Comment (alexcrichton) [inline src/cargo/ops/cargo_rustc/mod.rs]: This comment needs to move now that the conditional check moved

File: src/cargo/sources/registry/mod.rs [hunk 3]
```diff
@@ -322,10 +322,9 @@ impl<'cfg> Registry for RegistrySource<'cfg> {
         // theory the registry is known to contain this version. If, however, we
         // come back with no summaries, then our registry may need to be
         // updated, so we fall back to performing a lazy update.
-        if dep.source_id().precise().is_some() && !self.updated {
-            if try!(self.index.query(dep)).is_empty() {
+        if dep.source_id().precise().is_some() && !self.updated &&
+            try!(self.index.query(dep)).is_empty() {
                 try!(self.do_update());
```
Comment (alexcrichton) [inline src/cargo/sources/registry/mod.rs]: The indentation here is now wrong, can this stay as it was before? (more readable)

File: src/cargo/ops/cargo_rustc/mod.rs (hunk 2 above)
Comment (tshepang) [inline src/cargo/ops/cargo_rustc/mod.rs]: Comment looks awkward with my change, so I will leave it as it was.

**Comment by tshepang** [comment]
I failed to find what causes those test failures. I might try again if no one beats me to it.

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to resubmit with tests passing!

**Pull request closed without merging**